src/managers/k8s.py

```diff
@@ -42,28 +44,55 @@ def __init__(
             "SSL": "ssl",
         }
 
-    @cached_property
+    def __eq__(self, other: object) -> bool:
+        """__eq__ dunder.
+
+        Needed to get an cache hit on calls on the same method from different instances of K8sManager
+        as `self` is passed to methods.
+        """
+        return isinstance(other, K8sManager) and self.__dict__ == other.__dict__
+
+    def __hash__(self) -> int:
+        """__hash__ dunder.
+
+        K8sManager needs to be hashable so that `self` can be passed to the 'dict-like' cache.
+        """
+        return hash(json.dumps(self.__dict__, sort_keys=True))
```

Comment:
**thought, suggestion** maybe based on using private methods, you could also make the hidden methods static, taking `client` as an argument, in order to avoid this boiler plate

Reply:
`client` isn't hashable, so that wouldn't work I don't think.